src/components/CompareResults/ResultsView.tsx

```diff
@@ -34,6 +45,21 @@ function ResultsView(props: ResultsViewProps) {
     document.title = title;
   }, [title]);
 
+  useEffect(() => {
+    const waitForData = async () => {
+      await results;
+      setLoading(false);
+    };
+
+    void waitForData();
+  }, [results]);
```

Comment:
So this is working, but you know me, I'm not a big fan of effects ;-)

An alternative implementation would use `useLoaderData` and `<Await>` directly in `CancelAndCompareButton` to change the label, and get rid of all the `loading` props.

I wasn't sure this would be much better so I wrote down advantages and drawbacks as I see them:


Advantages:
* less passing props down
* code self-contained in CancelAndCompareButton, so there's no need to duplicate in "with base" and "over time"
* more tightly integrated within React rendering pipeline (with Suspense)

Drawbacks:
* CancelAndCompareButton more tightly coupled to the loader mechanism (initially I thought about passing a promise by prop but I think this is more complex)


In the end I think the advantages outnumber the drawbacks, so I'd rather do this rather than the loading boolean prop.
What do you think?

Reply:
We have to pass it down as a prop because the` CancelandCompareButton`component doesn't have access to the loader data on the Search Page so we can't use `useLoaderData` and since we have two loaders; it would be complicated.

Reply:
I've also removed any useEffects as well.

Reply:
Actually I found a work around with the search page and to access the loader only on the Results Page using Await and Suspense. What do you think?

```
  const renderCompareButton = () => {
    if (hasEditButton) {
      const { results } = useLoaderData() as LoaderReturnValue;
      return (
        <React.Suspense
          fallback={
            <Button id='compare-button' color='primary' type='submit' disabled>
              Loading...
            </Button>
          }
        >
          <Await resolve={results}>
            <Button id='compare-button' color='primary' type='submit'>
              {label}
            </Button>
          </Await>
        </React.Suspense>
      );
    } else
      return (
        <Button id='compare-button' color='primary' type='submit'>
          {label}
        </Button>
      );
  };
```

Reply:
"normal" react hooks can't be used in a if block, but I think the hooks from react router don't mind. So I don't mind either ;-)